pkg/cluster/reconcile.go

```diff
@@ -151,6 +151,29 @@ func (c *Cluster) removeOneMember() error {
 }
 
 func (c *Cluster) removeDeadMember(toRemove *etcdutil.Member) error {
+	if c.cluster.Spec.SelfHosted != nil {
+		selectedNodes, err := c.selectMasterNodes()
+		if err != nil {
+			return err
+		}
+
+		// If we do not have enough master nodes, we should simply wait for the old node
+		// to be online again.
+		//
+		// Removing the etcd member will not help us to recover the cluster to the desired
+		// number of members, since we do not have enough master nodes for self hosted etcd.
+		//
+		// Instead, there is a large chance that the old node is taken offline for maintenance
+		// or experiencing temporary network partition. When it comes back, it will recover itself
+		// since we persist data for self hosted case.
+
+		if nodeNum := len(selectedNodes); nodeNum < c.cluster.Spec.Size {
```

Comment:
So if I have 3 nodes, and expected size to be 3. What the code will do:
1. it will detect `nodeNum == spec.Size`
2. it will continue to delete the member